File: LeetCode/896.Monotonic_Array/main.py
#!/usr/bin/python
from solution import Solution

# Unit Test
v1 = [1,2,2,3]
v2 = [6,5,4,4]
v3 = [1,3,2]
v4 = [1,2,4,5]
v4 = [1,1,1,1]
s = Solution()
if (s.isMonotonic(v1)==True):
    print("PASS")
else:
    print("TEST 1 FAILED")

if (s.isMonotonic(v2)==False):
    print("PASS")
else:
    print("TEST 2 FAILED")


if (s.isMonotonic(v3)==False):
    print("PASS")
else:
    print("TEST 3 FAILED")


if (s.isMonotonic(v4)==False):
    print("PASS")
else:
    print("TEST 4 FAILED")

if (s.isMonotonic(v5)==False):
    print("PASS")
else:
    print("TEST 5 FAILED")

File: LeetCode/896.Monotonic_Array/main.cpp
#include "solution.h"

// Unit Test

int main(){
    Solution s;
    vector<int> v1 {1,2,2,3};
    vector<int> v2 {6,5,4,4};
    vector<int> v3 {1,3,2};
    vector<int> v4 {1,2,4,5};
    vector<int> v5 {1,1,1};
    
    if (s.isMonotonic(v1) == true) { cout << "Test 1 - PASS" << endl;}
    else { cout << "Test 1 - Failed" << endl; }
    if (s.isMonotonic(v2) == true) { cout << "Test 2 - PASS" << endl;}
    else { cout << "Test 2 - Failed" << endl; }
    if (s.isMonotonic(v3) == false) { cout << "Test 3 - PASS" << endl;}
    else { cout << "Test 3 - Failed" << endl; }
    if (s.isMonotonic(v4) == true) { cout << "Test 4 - PASS" << endl;}
    else { cout << "Test 4 - Failed" << endl; }
    if (s.isMonotonic(v5) == false) { cout << "Test 5 - PASS" << endl;}
    else { cout << "Test 5 - Failed" << endl; }
    return 0;
}